### DISCUSSION TITLE
Use Onnxruntime in C# with Unity for iOS: Can't load DLL 'onnxruntime'

### DISCUSSION THREAD

**Discussion by digitalmonkey** [microsoft/onnxruntime#14913] [Other Q&A]
I have a Unity App which uses the C# onnxruntime. It works fine on my MacOS device. When I build it for iOS, I created the Xcode file and manually added the iOS framework (I tried both the framework and xcframework) to the project. I'm able to compile and install on device and run the app. However, since onnxruntime is dynamically loaded, the moment my first onnxruntime code gets hit, I get the following error:

DllNotFoundException: Unable to load DLL 'onnxruntime'. Tried the load the following dynamic libraries: Unable to load dynamic library '/onnxruntime' because of 'Failed to open the requested dynamic library (0x06000000) dlerror() = dlopen(/onnxruntime, 0x0005): tried: '/onnxruntime' (errno=2), '/private/preboot/Cryptexes/OS/onnxruntime' (errno=2), '/onnxruntime' (errno=2)

It's a silent failure, in that it doesn't crash, none of my inference code is run either. I tried various ways of embedding the binary, but I just can't seem to open it. Anyone have any ideas?

**Comment by sharktide** [comment]
Similar issue with dotnet maui

**Comment by goekhan-qualcomm** [comment]
There is a repository that I tried for Unity once, which had interesting bits specifically for iOS. I never tried running things on iOS though. Nevertheless, here is the repo and the specific post processing step on how to include the runtime into the Xcode project:
https://github.com/asus4/onnxruntime-unity/blob/main/com.github.asus4.onnxruntime/Editor/OrtPostProcessBuild.cs

**Reply by goekhan-qualcomm** [reply]
@sharktide + @digitalmonkey... not sure if you ever fixed the issue.

**Reply by sharktide** [reply]
> @sharktide + @digitalmonkey... not sure if you ever fixed the issue.

I found a way to fix it temporarily with dotnet Maui projects by setting

``` 
<MTouchLink>SdkOnly</MTouchLink> 
``` 

but then a bunch of of other errors occur, but I'm pretty sure that is the right fix
